### PR TITLE
refmt fixer for ReasonML

### PR DESCRIPTION
**Pull request by ahmedelgabri**
Adding support for `refmt` as a fixer for ReasonML

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ formatting.
 | Puppet | [puppet](https://puppet.com), [puppet-lint](https://puppet-lint.com) |
 | Python | [autopep8](https://github.com/hhatto/autopep8), [flake8](http://flake8.pycqa.org/en/latest/), [isort](https://github.com/timothycrosley/isort), [mypy](http://mypy-lang.org/), [pycodestyle](https://github.com/PyCQA/pycodestyle), [pylint](https://www.pylint.org/) !!, [yapf](https://github.com/google/yapf) |
 | R | [lintr](https://github.com/jimhester/lintr) |
-| ReasonML | [merlin](https://github.com/the-lambda-church/merlin) see `:help ale-integration-reason-merlin` for configuration instructions |
+| ReasonML | [merlin](https://github.com/the-lambda-church/merlin) see `:help ale-integration-reason-merlin` for configuration instructions, [refmt](https://github.com/reasonml/reason-cli) |
```
Comment (w0rp) [inline README.md]: The table here needs to match the list in `doc/ale.txt`. Travis CI caught that.

File: autoload/ale/fixers/refmt.vim [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    return {
+    \   'command': ale#Escape(l:executable)
+    \       . (empty(l:options) ? '' : ' ' . l:options)
+    \       . ' --in-place'
+    \       . ' %t',
+    \   'read_temporary_file': 1,
+    \}
+endfunction
+
+
```
Comment (w0rp) [inline autoload/ale/fixers/refmt.vim]: There are some extra blank lines here to remove.

File: doc/ale-reasonml.txt [hunk 3]
```diff
@@ -10,6 +10,15 @@ merlin                                                    *ale-reasonml-merlin*
...
 
+===============================================================================
+ refmt                                                     *ale-reasonml-refmt*
+ 
+ g:ale_reasonml_refmt_options                    *g:ale_reasonml_refmt_options*
+                                                 *b:ale_reasonml_refmt_options*
+   Type: |String|
+   Default: `''`
+ 
+   This variable can be set to pass additional options to the refmt fixer.
```
Comment (w0rp) [inline doc/ale-reasonml.txt]: Don't forget to document the `_executable` option too.

File: README.md (hunk 1 above)
Comment (ahmedelgabri) [reply README.md]: I don't understand what you mean here?

**Comment by ahmedelgabri** [comment]
Everything is fixed now

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**